pandas/core/groupby.py

```diff
@@ -2883,7 +2883,12 @@ def aggregate(self, func_or_funcs, *args, **kwargs):
             except Exception:
                 result = self._aggregate_named(func_or_funcs, *args, **kwargs)
 
-            index = Index(sorted(result), name=self.grouper.names[0])
+            # mixed types fail to sort
+            try:
+                values = sorted(result)
+            except TypeError:
+                values = result
```

Comment:
yeah this is an embedded issue with ``safe_sort``, fixed here: https://github.com/pandas-dev/pandas/pull/17034. you can rebase on top (or wait for me to merge this).